core-bundle/src/Resources/contao/pages/PageError401.php

```diff
@@ -122,7 +122,7 @@ protected function prepare($objRootPage=null)
 			}
 
 			// Add the referer so the login module can redirect back
-			$url = $objNextPage->getAbsoluteUrl() . '?redirect=' . Environment::get('base') . Environment::get('request');
+			$url = $objNextPage->getAbsoluteUrl() . '?redirect=' . urlencode(Environment::get('base') . Environment::get('request'));
```

Comment:
Are you sure that `urldecode()` instead of `rawurldecode()` is correct here?

Reply:
```suggestion
			$url = $objNextPage->getAbsoluteUrl() . '?redirect=' . rawurlencode(Environment::get('base') . Environment::get('request'));
```